### PR TITLE
Link to Philips Hue release notes in vendor link

### PR DESCRIPTION
**Pull request by davidjb**
This links to the Philips Hue release notes URL and also fixes a minor typo in the filename for the VendorLinks component.

### CODE REVIEW

File: src/components/vendor-links/vendor-links.tsx [hunk 1]
```diff
@@ -49,6 +49,10 @@ export const OTALink: React.FunctionComponent<VendorProps> = (props: VendorProps
             url = `https://ww8.ikea.com/ikeahomesmart/releasenotes/releasenotes.html`
             break
 
+        case "Philips":
+            url = `https://www.philips-hue.com/en-us/support/release-notes`
```
Comment (nurikk) [inline src/components/vendor-links/vendor-links.tsx]: Can we try to guess what type of device is this using `device.definition` if yes, we can land users to either
`https://www.philips-hue.com/en-us/support/release-notes/accessories` or `https://www.philips-hue.com/en-us/support/release-notes/lamps`

Comment (davidjb) [reply src/components/vendor-links/vendor-links.tsx]: @nurikk Great suggestion - I've done this by checking `device.definition?.exposes` for a `light` capability and linking to lamps if found, and to accessories for everything else.

**Comment by davidjb** [comment]
This now links to `/lamps` for lights and to `/accessories` for everything else.

In order to do this, I had to adjust the type for `DeviceDefinition` as the previous definition prevented calling array methods on `.exposes`.  While I was there, I applied the same fix to `.options` and simplified `CompositeFeature.features` too.

**Comment by nurikk** [comment]
Nicely done! Thanks 🙏

**Pull request merged**